numpy/core/tests/test_multiarray.py

```diff
@@ -1357,6 +1357,17 @@ def test_multiindex_titles(self):
         assert_raises(ValueError, lambda : a[['b','b']])  # field exists, but repeated
         a[['b','c']]  # no exception
 
+    def test_structured_cast_fieldorder(self):
+        # gh-15494
+        A = ("a", "<i8")
+        B = ("b", "<i8")
+        ab = np.rec.array(obj=np.array([(1, 2), dtype='<i8']), dtype=[A, B])
```

Comment:
Let's not throw `recarray` in here, the issue is more fundamental:
```suggestion
        ab = np.array([(1, 2)], dtype=[A, B])
```